go-controller/pkg/ovn/master.go

```diff
@@ -228,6 +241,159 @@ func (oc *Controller) syncNodeManagementPort(node *kapi.Node) error {
 	return nil
 }
 
+func parseGatewayIfaceID(node *kapi.Node) (string, error) {
+	ifaceID, ok := node.Annotations[OvnNodeGatewayIfaceID]
+	if !ok {
+		logrus.Errorf("gateway ifaceID annotation not found for node %q ", node.Name)
+		return "", nil
+	}
+
+	return ifaceID, nil
+}
+
+func parseGatewayMacAddress(node *kapi.Node) (string, error) {
+	gatewayMacAddress, ok := node.Annotations[OvnNodeGatewayMacAddress]
+	if !ok {
+		logrus.Errorf("gatewayMacAddress annotation not found for node %q ", node.Name)
+		return "", nil
+	}
+
+	_, err := net.ParseMAC(gatewayMacAddress)
+	if err != nil {
+		return "", fmt.Errorf("Error %v in parsing node gateway macAddress %v", err, gatewayMacAddress)
+	}
+
+	return gatewayMacAddress, nil
+}
+
+func parseGatewayLogicalNetwork(node *kapi.Node) (string, string, error) {
+	ipAddress, ok := node.Annotations[OvnNodeGatewayIP]
+	if !ok {
+		logrus.Errorf("gatewayIpAddress annotation not found for node %q ", node.Name)
+		return "", "", nil
+	}
+
+	gwNextHop, ok := node.Annotations[OvnNodeGatewayNextHop]
+	if !ok {
+		logrus.Errorf("gatewayNextHop annotation not found for node %q ", node.Name)
+		return "", "", nil
+	}
+
+	return ipAddress, gwNextHop, nil
+}
+
+func (oc *Controller) syncGatewayLogicalNetwork(node *kapi.Node, mode string) error {
+
+	if mode == string(config.GatewayModeDisabled) {
+		return nil
+	}
+
+	var cidr string
+	var subnet *net.IPNet
+	var err error
+
+	if cidr, _, err = util.RunOVNNbctl("get", "logical_switch", node.Name, "other-config:subnet"); err != nil {
+		return err
+	}
+	if cidr == "" {
+		return fmt.Errorf("Failed to obtain gateway subnet for clean up node %s", node.Name)
+	}
+
+	_, subnet, err = net.ParseCIDR(cidr)
+	if err != nil {
+		logrus.Errorf("Invalid hostsubnet found for node %s - %v", node.Name, err)
+		return err
+	}
+
+	var clusterSubnets []string
+	for _, clusterSubnet := range config.Default.ClusterSubnets {
+		clusterSubnets = append(clusterSubnets, clusterSubnet.CIDR.String())
+	}
+	if len(clusterSubnets) == 0 {
+		return nil
+	}
+
```

Comment:
L312-314: can be removed since it is guaranteed to be non-Zero. If this is not set, then we have bigger issues. Furthermore, you are returning `nil`, so that is not good too. So, it is better to remove it :-)